tests/test_modeling_bigbird_pegasus.py

```diff
@@ -0,0 +1,700 @@
+
+        self.parent.assertTrue((encoder_last_hidden_state_2 - encoder_last_hidden_state).abs().max().item() < 1e-3)
+
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            decoder = model.get_decoder()
+            decoder.save_pretrained(tmpdirname)
+            decoder = BigBirdPegasusDecoder.from_pretrained(tmpdirname).to(torch_device)
+
+        last_hidden_state_2 = decoder(
+            input_ids=inputs_dict["decoder_input_ids"],
+            attention_mask=inputs_dict["decoder_attention_mask"],
+            encoder_hidden_states=encoder_last_hidden_state,
+            encoder_attention_mask=inputs_dict["attention_mask"],
+        )[0]
+
+        self.parent.assertTrue((last_hidden_state_2 - last_hidden_state).abs().max().item() < 1e-3)
+
+    def create_and_check_model(self, config, inputs_dict):
+        model = BigBirdPegasusModel(config=config).to(torch_device).eval()
+        input_ids = inputs_dict["input_ids"]
+        decoder_input_ids = inputs_dict["decoder_input_ids"]
+        result = model(input_ids, decoder_input_ids=decoder_input_ids, use_cache=True)
+        self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.seq_length, self.hidden_size))
+
+    # def create_and_check_for_auto_padding(
+    #     self,
+    #     config,
+    #     input_ids,
+    #     token_type_ids,
+    #     input_mask,
+    #     sequence_labels,
+    #     token_labels,
+    #     choice_labels,
+    # ):
+    #     model = BigBirdModel(config)
+    #     model.to(torch_device)
+    #     model.eval()
+    #     result = model(input_ids)
+    #     self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.seq_length, self.hidden_size))
+
+    # def create_and_check_for_change_to_full_attn(
+    #     self,
+    #     config,
+    #     input_ids,
+    #     token_type_ids,
+    #     input_mask,
+    #     sequence_labels,
+    #     token_labels,
+    #     choice_labels,
+    # ):
+    #     model = BigBirdModel(config)
+    #     model.to(torch_device)
+    #     model.eval()
+    #     result = model(input_ids)
+    #     self.parent.assertEqual(result.last_hidden_state.shape, (self.batch_size, self.seq_length, self.hidden_size))
+    #     # the config should not be changed
+    #     self.parent.assertTrue(model.config.attention_type == "block_sparse")
+
+
+@require_torch
+class BigBirdPegasusModelTest(ModelTesterMixin, GenerationTesterMixin, unittest.TestCase):
+    all_model_classes = (
+        (
+            BigBirdPegasusModel,
+            BigBirdPegasusForConditionalGeneration,
+            BigBirdPegasusForSequenceClassification,
+            BigBirdPegasusForQuestionAnswering,
+        )
+        if is_torch_available()
+        else ()
+    )
+    all_generative_model_classes = (BigBirdPegasusForConditionalGeneration,) if is_torch_available() else ()
+    is_encoder_decoder = True
+    test_pruning = False
+    test_head_masking = False
+    test_missing_keys = False
```

Comment:
this I think we can set to True to test it no?

Reply:
@patrickvonplaten can you explain a bit about what this test is doing?? This test is failing currently, since `len(missing_keys) != 0`, but shouldn't it be 0??